delorean/interface.py

```diff
@@ -10,23 +10,47 @@
 UTC = "UTC"
 utc = timezone("utc")
 
+NAIVEMODE_UTC = "utc"
+NAIVEMODE_NAIVE = "naive"
+NAIVEMODE_RAISE = "raise"
+NAIVE_MODES = (NAIVEMODE_UTC, NAIVEMODE_NAIVE, NAIVEMODE_RAISE)
 
-def parse(s, dayfirst=True, yearfirst=True):
+def parse(s, dayfirst=True, yearfirst=True, naivemode=NAIVEMODE_UTC):
     """
     Parses a datetime string in it and returns a `Delorean` object.
 
     If a timezone is detected in the datetime string it will be
     normalized to UTC, and a Delorean object with that datetime and
     timezone will be returned.
+
+    If a timezone is not detected, the resulting timezone (and return object)
+    depends on the value of the naivemode argument:
+        '{mode_utc}' -- UTC is assumed
+        '{mode_raise}' -- An exception is raised
+        '{mode_naive}' -- Returns a naive datetime object (NOT a `Delorean`).
+
+    Note that in the case of naivemode=='{mode_naive}' a Delorean object is
+    not returned. This is because Delorean objects should always be timezone
+    aware.
+
     """
+
+    if not naivemode in NAIVE_MODES:
+        raise ValueError("Unknown naivemode value: '%s'" % naivemode)
+
     try:
         dt = capture(s, dayfirst=dayfirst, yearfirst=yearfirst)
     except:
         # raise a parsing error.
         raise ValueError("Unknown string format")
     if dt.tzinfo is None:
-        # assuming datetime object passed in is UTC
-        do = Delorean(datetime=dt, timezone=UTC)
+        if naivemode == NAIVEMODE_UTC:
+            # assuming datetime object passed in is UTC
+            do = Delorean(datetime=dt, timezone=UTC)
+        elif naivemode == NAIVEMODE_RAISE:
+            raise DeloreanInvalidDatetime("Unable to determine timezone info")
+        elif naivemode == NAIVEMODE_NAIVE:
+            do = dt #note that we are *not* returning a Delorean here
```

Comment:
I don't particularly like the smell of that either.  I originally started with returning a `Delorean` where the tz was None.  I think it actually passed all unit tests, but figured that that was not legit.  My guess is that having a legit tz is an expectation engrained in all of delorean, and having None would break a lot of ops.  Maybe not?

Maybe an alternative is to drop `naivemode='naive'` and *always* return a `Delorean`, but still have the `naivemode='raise'` option.  The naive `datetime` could be included in the raised exception.  Returning usable data in an exception smells even worse, though.  If just an exception, a subsequent call could then be made to a (new) `delorean.parse_datetime` call which always returns a `datetime`, but that seems like busy work to have to call twice.

Or maybe a whole new call that returns both a `Delorean` representation of the time *and* a datetime representation?

    >>> ret = delorean.parse2('2015-02-04T16:33:21.247513Z') #Z for UTC
    >>> [type(v) for v in ret]
    [<class 'delorean.dates.Delorean'>, <type 'datetime.datetime'>]
    >>> ret = delorean.parse2('2015-02-04T16:33:21.247513Z')  # no tz data
    [<type 'NoneType'>, <type 'datetime.datetime'>]

where the Delorean ret is None when there's no tz data.

That seems overly complex, though.  My current code is simpler, although the return type *is* dependent on an arg value.